### ISSUE TITLE
transferArgumentDocumentation is erroneous

### ISSUE THREAD

**Issue by tolnaisz** [doxygen/doxygen#7943]
The transferArgumentDocumentation as it is implemented at the moment is pretty much a no-op. The issue is that the decA and defA are not references, which means that any modification onto those will not be reflected in the input parameters decAl and defAl as expected.

**Comment by albert-github** [comment]
I assume you mean the function `transferArgumentDocumentation` in `memeberdef.cpp`.

Can you give an example where it shows up that it has no effect as my interpretation is that the documentation from `defA`  is copied to the documentation of `decA` when the documentation of `defA` is not empty and the documentation of `decA` is empty.

**Comment by tolnaisz** [comment]
Yes, I meant the function ```transferArgumentDocumentation``` in ```memberdef.cpp```. The code creates a copy of ```Argument``` and not a reference to it. The code then, as you pointed that out, copies the documentation from one to the other correctly. However, this happens on the local copies of the ```Argument```s and not the ones inside the ```ArgumentList```s. It would be enough to simply change the local copies to be references instead. I.e.:
```
    Argument& decA = *decIt;
    Argument& defA = *defIt;

```

**Comment by doxygen** [comment]
@tolnaisz That's not good indeed. Well spotted.

**Comment by albert-github** [comment]
A bit weird is that nobody complained about a problem (in the output) as this fix was introduced by:
```
Commit: c38a6fe88ceb3658ae4c09fa9550c770195d9401 [c38a6fe]
Date: Sunday, October 27, 2019 12:55:16 PM
Changed ArgumentList to be an STL container
```
and was in version 1.8.17 (December 2019).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
